### PR TITLE
Add support for the cold_standby state

### PR DESCRIPTION
**Pull request by TuxPowered42**

### CODE REVIEW

File: src/check_libvirt_hosts.py [hunk 1]
```diff
@@ -197,17 +198,37 @@ def check_extraneous_domains(domains, vms):
...
 
-def check_inactive_domains(domains):
-    #  - Check if there are VMs in libvirt in a non-running state
+def check_inactive_domains(domains, vms):
+    # Check if there are domains in libvirt in a non-running state, apart from ones
+    # that are marked as non-running in Serveradmin.
     inactive_domains = set(
-        d['hostname'] for d in domains.values() if not d['is_active'])
+        d['hostname'] for d_id, d in domains.items() if not d['is_active'] and
+        vms.get(d_id, {}).get('state') not in ('cold_standby', 'retired')
```
Comment (lamaral) [inline src/check_libvirt_hosts.py:206]:
```suggestion
        vms.get(d_id, {}).get('state') not in ['cold_standby', 'retired']
```

Comment (TuxPowered42) [reply src/check_libvirt_hosts.py:206]: Using tuples for such cases is fine. And for many cases using tuples is preferred over arrays when memory and cpu usage are important.

File: src/check_libvirt_hosts.py [hunk 2]
```diff
@@ -197,17 +198,37 @@ def check_extraneous_domains(domains, vms):
...
+    )
     if len(inactive_domains) > 0:
         code = ExitCodes.warning
-        message = 'Found non-running domains:'
+        message = 'Found non-running domains which should be running:'
         message += indent_hostname_list(inactive_domains)
     else:
         code = ExitCodes.ok
-        message = 'All domains are running'
+        message = 'Configured as running domains are running'
```
[the remark this answers is not in the document]
Comment (TuxPowered42) [reply src/check_libvirt_hosts.py]: Fixed.

File: src/check_libvirt_hosts.py [hunk 3]
```diff
@@ -197,17 +198,37 @@ def check_extraneous_domains(domains, vms):
...
+        d['hostname'] for d_id, d in domains.items() if d['is_active'] and
+        vms.get(d_id, {}).get('state') in ('cold_standby', 'retired')
+    )
+    if len(standby_domains) > 0:
+        code = ExitCodes.warning
+        message = 'Found running domains which should not be running:'
+        message += indent_hostname_list(standby_domains)
+    else:
+        code = ExitCodes.ok
+        message = 'Configured as non-running domains are powered off'
```
[the remark this answers is not in the document]
Comment (TuxPowered42) [reply src/check_libvirt_hosts.py]: Fixed.

**Review by lamaral — APPROVED** [review]

**Pull request merged**